design_docs/generalize_calendar.md

```diff
@@ -0,0 +1,249 @@
+manipulation of dates is required.
+
+A successful implementation would produce essentially identical analysis to
+what is currently produced, but making use of the dates from the MPAS calendar
+(whether Gregorian or 365-day) without the need for artifical offsets (e.g.
+`yearOffset` used in the current code.  Plots of horizontal fields would remain
+unchanged while plots of time series would have a time axis with the simulation
+date instead of the offset date.
+
+
+<h1> Requirements </h1>
+
+<h2> Requirement: The 'Time' coordinate of xarray data sets must be consistent 
+with the MPAS calendar <br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+For all data sets used in the analysis, the 'Time' coordinate must represent dates 
+on the appropriate MPAS calendar, either 'gregorain' or 'gregorain_noleap', depending
+on the namelist option 'config_calendar_type'.  There must be ways of mathematically
+manipulating times (e.g. adding/subtracting offsets and figuring out the amount of time
+between two dates) and of making plots that are consistent with these calendars.
+
+<h2> Requirement: The 'Time' coordinate of xarray data sets must support years 
+0001 and 9999, and value, both positive and negative<br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+For all data sets used in the analysis, the 'Time' coordinate must, at a minimum, 
+support years between 0001 and 9999 (the range of `datetime.datetim`) and preferably
+a broader range.
+
+
+<h1> Algorithmic Formulations (optional) </h1>
+
+<h2> Design solution: The 'Time' coordinate of xarray data sets must be consistent
+with the MPAS calendar <br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The proposed solution represents time in `xarray.DataSet` objects as the number of
+days since the reference date 0001-01-01.  For purposes of performing mathematical
+operations and plotting dates, these values will be converted to 
+`datetime.datetime` objects (via the proposed `daysToDatetime` utility function)
+and back (via the proposed `datetimeToDays`).
+
+The conversion operations will be performed with the calendar-aware functions
+`netCDF4.date2num` and `netCDF4.num2date` and will support lists/arrays of dates
+(for efficiency and simplicity of calling code) in addition to single values.
+
+<h2> Design solution: The 'Time' coordinate of xarray data sets must support years
+0001 and 9999, and value, both positive and negative<br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+Same as above. In theory, the use of days since 0001-01-01 would allow any year
+to be supported, not just the range from 0001 to 9999.  However, the conversions
+to `datetime.datetime` objects for mathematical manipulation will constrain
+the dates to be between datetime.min (0001-01-01) and datetime.max (9999-12-31).
+
+
+<h1> Design and Implementation </h1>
+
```

Comment:
Yep, I've got an implementation already that needs cleaning up based on various other PRs in progress.  The branch exists if you want to take a look but I haven't make a PR because it's not ready for review yet.